platform/linux-generic/include/odp_schedule_scalable_ordered.h

```diff
@@ -79,6 +79,7 @@ typedef struct reorder_window {
 	uint32_t tail;
 	uint32_t turn;
 	uint32_t olock[CONFIG_QUEUE_MAX_ORD_LOCKS];
+	uint32_t lock_index;
```

Comment:
Why is this uint32_t? The lock_count (defined below) is only an uint16_t. We don't support more than 65535 locks (per ordered queue). The ordered locks themselves use uint32_t (for the "ticket") but also that is overkill, there just won't be that many threads.

Reply:
The lock_index is an implied parameter to `odp_schedule_order_unlock_lock()`. It's the index of the lock that the caller currently holds, so this is a thread-local variable. It's an error to call this API without already holding an ordered lock.

Reply:
Not true. Linux-generic implementation is using an uint16_t but as explained before implementations are free to use any number of bits as required. The number of bits of lock_index also depends upon the application use-case.
Ordered locks is not related to number of threads in the system but to the total number of flows in the system. Depending upon the implementation it could support a huge number of flows ( 2^20 ) and not all flows will be active.